core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2515,21 +2508,15 @@ else if (expression instanceof DereferenceExpression) {
                         name = DereferenceExpression.getQualifiedName((DereferenceExpression) expression);
                     }
 
-                    if (name != null) {
-                        List<Field> matchingFields = sourceScope.getRelationType().resolveFields(name);
-                        if (!matchingFields.isEmpty()) {
-                            originTable = matchingFields.get(0).getOriginTable();
-                            originColumn = matchingFields.get(0).getOriginColumnName();
-                        }
-                    }
+                    builder.addAll(analysis.getColumnOriginDetails(expression));
```

Comment:
If `expression`  is a complex expression, it might reference many fields, and you'll get all of them.
I think you need to surround this call with `if (name != null)`, which limits the case to resolving a single field. In such case there should be one `details`.

Reply:
> it might reference many fields, and you'll get all of them.

We require the list of dependent field/column